k2/csrc/macros.h

```diff
@@ -21,4 +21,16 @@
 #define K2_FUNC __func__
 #endif
 
+#define K2_EVAL(context, dim, lambda_name, arg_name, ...)              \
```

Comment:
Need some documentation for this macro, please!

Reply:
... also please try using it at least once.
Eventually we'll have a K2_EVAL2 macro, for 2-arg lambdas.

Reply:
Documentation is added.

`K2_EVAL2` is also added.

I've replaced nearly all ocurrences of Eval and Eval2 with K2_EVAL and K2_EVAL2.